src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -310,19 +312,14 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             return method;
         }
 
-        //TODO See if SSL_CTX_Set_quite_shutdown can be used
         private static void Disconnect(IntPtr sslPtr)
         {
             if (IntPtr.Zero != sslPtr)
             {
                 int retVal = libssl.SSL_shutdown(sslPtr);
-                if (0 == retVal)
+                if (retVal < 0)
                 {
-                    retVal = libssl.SSL_shutdown(sslPtr);
-                    if (retVal < 0)
-                    {
-                        libssl.SSL_get_error(sslPtr, retVal);
-                    }
+                    libssl.SSL_get_error(sslPtr, retVal);
```

Comment:
It's not always documented which operations do or do not push errors into the queue, so we check for any operation that might.  That means that if an error is left in the queue, we may end up picking it up when querying for a completely unrelated operation.  At least that's how things were a while ago... @bartonjs can correct me if things have changed.